src/main/java/com/autotune/analyzer/kruizeObject/KruizeObject.java

```diff
@@ -133,24 +136,73 @@ public KruizeObject() {
+        }
+        else {
+            // for local monitoring use case
+            // Default is Short Term
+            terms.put(KruizeConstants.JSONKeys.SHORT_TERM, new Terms(KruizeConstants.JSONKeys.SHORT_TERM, KruizeConstants.RecommendationEngineConstants
+                    .DurationBasedEngine.DurationAmount.SHORT_TERM_DURATION_DAYS, KruizeConstants.RecommendationEngineConstants
+                    .DurationBasedEngine.DurationAmount.SHORT_TERM_DURATION_DAYS_THRESHOLD, 4, 0.25));
+
+            kruizeObject.setTerms(terms);
+        }
+    }
+
+
+    public static void setCustomTerms(Map<String, Terms> terms, KruizeObject kruizeObject) throws InvalidTermException {
+        // TODO: define term names like daily, weekly, fortnightly etc
+        // TODO: add CRD for terms
 
-        kruizeObject.setTerms(terms);
+
+        if(  kruizeObject.getRecommendation_settings() != null &&
+                kruizeObject.getRecommendation_settings().getTermSettings() != null )
+        {
+
+            String userInputTerm = String.valueOf(kruizeObject.getRecommendation_settings().getTermSettings().getTerms().get(0));
+
+            if( "short".equalsIgnoreCase(userInputTerm)){
+                terms.put(KruizeConstants.JSONKeys.SHORT_TERM, new Terms(KruizeConstants.JSONKeys.SHORT_TERM, KruizeConstants.RecommendationEngineConstants
+                        .DurationBasedEngine.DurationAmount.SHORT_TERM_DURATION_DAYS, KruizeConstants.RecommendationEngineConstants
+                        .DurationBasedEngine.DurationAmount.SHORT_TERM_DURATION_DAYS_THRESHOLD, 4, 0.25));
+            }
+            else if ( "medium".equalsIgnoreCase(userInputTerm)) {
+                terms.put(KruizeConstants.JSONKeys.MEDIUM_TERM, new Terms(KruizeConstants.JSONKeys.MEDIUM_TERM, KruizeConstants
+                        .RecommendationEngineConstants.DurationBasedEngine.DurationAmount.MEDIUM_TERM_DURATION_DAYS, KruizeConstants
+                        .RecommendationEngineConstants.DurationBasedEngine.DurationAmount.MEDIUM_TERM_DURATION_DAYS_THRESHOLD, 7, 1));
+            }
+            else if ( "long".equalsIgnoreCase(userInputTerm)) {
+                terms.put(KruizeConstants.JSONKeys.LONG_TERM, new Terms(KruizeConstants.JSONKeys.LONG_TERM, KruizeConstants
+                        .RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS, KruizeConstants
+                        .RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS_THRESHOLD, 15, 1));
+
+            }
+           else{
+               // Term name is invalid
+               throw new InvalidTermException(userInputTerm + " is not supported, use short, medium or long term");
+           }
+            kruizeObject.setTerms(terms);
+        }
+        else {
+            // Handle the case where termSettings is null
+            throw new InvalidTermException("Term settings are not defined in the recommendation settings.");
```

Comment:
Please use constants for all strings and move them to [here](https://github.com/kruize/autotune/blob/master/src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java)